sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -80,19 +88,23 @@ class CloudEvent(EventMixin):   #pylint:disable=too-many-instance-attributes
      unique for each distinct event. If not provided, a random UUID will be generated and used.
     :vartype id: Optional[str]
     """
-    def __init__(self, source, type, data, **kwargs): # pylint: disable=redefined-builtin
-        # type: (str, str, object, Any) -> None
+    def __init__(self, source, type, **kwargs): # pylint: disable=redefined-builtin
+        # type: (str, str, Any) -> None
         self.source = source
         self.type = type
         self.specversion = kwargs.pop("specversion", "1.0")
         self.id = kwargs.pop("id", str(uuid.uuid4()))
         self.time = kwargs.pop("time", dt.datetime.now(UTC()).isoformat())
-        self.data = data
+        self.data = kwargs.pop("data", None)
         self.datacontenttype = kwargs.pop("datacontenttype", None)
         self.dataschema = kwargs.pop("dataschema", None)
         self.subject = kwargs.pop("subject", None)
+        self.data_base64 = kwargs.pop("data_base64", None)
         self.extensions = {}
         self.extensions.update(dict(kwargs.pop('extensions', {})))
+        if self.data is not None and self.data_base64 is not None:
+            raise ValueError("data and data_base64 cannot be provided at the same time.\
+                Use data_base64 only if you are sending bytes, and use data otherwise.")
```

Comment:
do we need to check if both of them are None?
```python
if not self.data and not self.data_base64:
    raiseValueError("...")
```

Reply:
any do we need to check if `data_base64` is of bytes type?

Reply:
yes - data can totally be an empty dict for instance - so we need to check if both are not none explicitly.

for second part, not necessarily, we will eventually get a good 
`TypeError: a bytes-like object is required, not 'str'`